euphony/src/main/cpp/tests/utf16CharsetTest.cpp

```diff
@@ -0,0 +1,65 @@
+#include <gtest/gtest.h>
+#include <Definitions.h>
+#include <UTF16Charset.h>
+#include <tuple>
+
+using namespace Euphony;
+
+typedef std::tuple<std::string, std::string> TestParamType;
+
+class UTF16CharsetTestFixture : public ::testing::TestWithParam<TestParamType>
+{
+
+public:
+    void openCharset()
+    {
+        EXPECT_EQ(charset, nullptr);
+        charset = new UTF16Charset();
+        ASSERT_NE(charset, nullptr);
+    }
+
+    Charset *charset = nullptr;
+};
+
+TEST_P(UTF16CharsetTestFixture, EncodingTest)
+{
+openCharset();
+
+std::string source;
+std::string expectedResult;
+
+std::tie(source, expectedResult) = GetParam();
+
+HexVector actualResult = charset->encode(source);
+EXPECT_EQ(actualResult.toString(), expectedResult);
+}
+
+TEST_P(UTF16CharsetTestFixture, DecodingTest)
+{
+openCharset();
+
+std::string source;
+std::string expectedResult;
+
+std::tie(expectedResult, source) = GetParam();
+HexVector hv = HexVector(source);
+
+std::string actualResult = charset->decode(hv);
+EXPECT_EQ(actualResult, expectedResult);
+}
```

Comment:
No indentation applied here.
Would you please apply indentation to here?

Reply:
@kuro11pow2 @seoh02h 
Thank you for your comment! We will revise the indentation of the code and modify it to K&R style as well.

Reply:
@kuro11pow2 @seoh02h Thanks to fix indentation :)